charms/pipelines-ui/src/charm.py

```diff
@@ -93,6 +93,44 @@ def set_pod_spec(self, event):
 
         healthz = f"http://localhost:{config['http-port']}/apis/v1beta1/healthz"
 
+        env = {
+            "ALLOW_CUSTOM_VISUALIZATIONS": str(
+                config["allow-custom-visualizations"]
+            ).lower(),
+            "ARGO_ARCHIVE_ARTIFACTORY": "minio",
+            "ARGO_ARCHIVE_BUCKETNAME": "mlpipeline",
+            "ARGO_ARCHIVE_LOGS": "false",
+            "ARGO_ARCHIVE_PREFIX": "logs",
+            "AWS_ACCESS_KEY_ID": "",
+            "AWS_SECRET_ACCESS_KEY": "",
+            "DEPLOYMENT": "KUBEFLOW",
+            "DISABLE_GKE_METADATA": "false",
+            "ENABLE_AUTHZ": "false",
+            "HIDE_SIDENAV": "",
+            "HTTP_AUTHORIZATION_DEFAULT_VALUE": "",
+            "HTTP_AUTHORIZATION_KEY": "",
+            "HTTP_BASE_URL": "",
+            "KUBEFLOW_USERID_HEADER": "kubeflow-userid",
+            "KUBEFLOW_USERID_PREFIX": "",
+            "METADATA_ENVOY_SERVICE_SERVICE_HOST": "localhost",
+            "METADATA_ENVOY_SERVICE_SERVICE_PORT": "9090",
+            "MINIO_ACCESS_KEY": minio["user"],
+            "MINIO_HOST": minio["ip"],
+            "MINIO_NAMESPACE": self.model.name,
+            "MINIO_PORT": minio["port"],
+            "MINIO_SECRET_KEY": minio["password"],
+            "MINIO_SSL": "false",
+            "ML_PIPELINE_SERVICE_HOST": api["service"],
+            "ML_PIPELINE_SERVICE_PORT": api["port"],
+            "STREAM_LOGS_FROM_SERVER_API": "false",
+            "VIEWER_TENSORBOARD_POD_TEMPLATE_SPEC_PATH": "/etc/config/viewer-pod-template.json",
+            "VIEWER_TENSORBOARD_TF_IMAGE_NAME": "tensorflow/tensorflow",
+        }
+
+        config_json = json.dumps(
+            {"spec": {"serviceAccountName": "kubeflow-pipelines-viewer"}}
+        )
+
```

Comment:
Code comments on lines which are not part of the commit in this file:

- L26: `relations` seems like weird name for this if it also includes the basic events
- L44: `send_info` might be a bit vague maybe `send_info_to_pipelines`?
- L59-L75: This should use the [minio_interface library](https://github.com/canonical/minio-operator/blob/master/lib/charms/minio/v0/minio_interface.py) in the minio charm.  See the [argo-controller](https://github.com/canonical/argo-operators/tree/main/operators/argo-controller) charm for an example